### DISCUSSION TITLE
Data migration tool to import data into Marten

### DISCUSSION THREAD

**Discussion by mysticmind** [JasperFx/marten#944] [Ideas]
Provide a generic data migration tool to ease the process/reduce friction to import data into Marten from other data sources such as SQL Server, CSV etc.

CosmosDB team has built a tool which provides both GUI and commandline tools to import data from JSON files, MongoDB, SQL Server, CSV files, Azure Table Storage, Amazon DynamoDB, HBase, Azure Cosmos DB collections. [Source code](https://github.com/azure/azure-documentdb-datamigrationtool) is available in GitHub  and MIT licensed. You can refer to documentation [here](https://docs.microsoft.com/en-us/azure/cosmos-db/import-data). This helps to pick some ideas to build a data migration tool covering several data sources.

**Comment by mysticmind** [comment]
Closing this, we can revisit this enhancement later.

**Comment by oskardudycz** [comment]
@mysticmind I think that maybe we could keep it open to not forget about that? I see this really useful, especially for migration from the relational databases (so help potentially migration from EF).

**Comment by mysticmind** [comment]
Okay, re-opening this issue
